crates/core/tests/client.rs

```diff
@@ -1,20 +1,145 @@
+mod helpers;
+
 #[cfg(test)]
 mod tests {
 
     use std::str::FromStr;
 
+    use dojo_test_utils::sequencer::TestSequencer;
+    use ethers::types::{Address as EthersAddress, U256 as EthersU256};
     use kakarot_rpc_core::client::api::{KakarotEthApi, KakarotStarknetApi};
+    use kakarot_rpc_core::client::config::StarknetConfig;
     use kakarot_rpc_core::client::errors::EthApiError;
+    use kakarot_rpc_core::client::KakarotClient;
     use kakarot_rpc_core::mock::wiremock_utils::setup_mock_client_crate;
     use kakarot_rpc_core::models::block::BlockWithTxs;
     use kakarot_rpc_core::models::convertible::{ConvertibleStarknetBlock, ConvertibleStarknetEvent};
     use kakarot_rpc_core::models::event::StarknetEvent;
+    use kakarot_rpc_core::models::felt::Felt252Wrapper;
     use kakarot_rpc_core::models::ConversionError;
     use reth_primitives::{Address, Bytes, H256};
     use reth_rpc_types::Log;
     use starknet::core::types::{BlockId, BlockTag, Event, FieldElement};
     use starknet::core::utils::get_selector_from_name;
-    use starknet::providers::Provider;
+    use starknet::providers::jsonrpc::HttpTransport;
+    use starknet::providers::{JsonRpcClient, Provider};
+
+    use crate::helpers::constants::EOA_WALLET;
+    use crate::helpers::deploy_helpers::{create_raw_ethereum_tx, deploy_kakarot_system};
+
+    #[tokio::test]
+    async fn test_constructable() {
+        // initial setup of Constructable to test we can deploy contracts w/ constructor arguments
+        let starknet_test_sequencer = TestSequencer::start().await;
+
+        let expected_funded_amount = FieldElement::from_dec_str("100000000000000000").unwrap();
+
+        let deployed_kakarot =
+            deploy_kakarot_system(&starknet_test_sequencer, EOA_WALLET.clone(), expected_funded_amount).await;
```

Comment:
just want to leave a note about why I used `clone` here, as I am beginning to understand Rust's memory management model

just trying to pass `EOA_WALLET` gives me

```
   --> crates/core/tests/client.rs:38:61

    |

38  |             deploy_kakarot_system(&starknet_test_sequencer, EOA_WALLET, expected_funded_amount).await;

    |             ---------------------                           ^^^^^^^^^^ expected `Wallet<SigningKey<Secp256k1>>`, found `EOA_WALLET`

    |             |

    |             arguments to this function are incorrect

    |

    = note: expected struct `Wallet<ecdsa::signing::SigningKey<Secp256k1>>`

               found struct `EOA_WALLET`

```

so then I would want to derefence `*EOA_WALLET` and see:

```
error[E0507]: cannot move out of dereference of `EOA_WALLET`

  --> crates/core/tests/client.rs:38:61

   |

38 |             deploy_kakarot_system(&starknet_test_sequencer, *EOA_WALLET, expected_funded_amount).await;

   |                                                             ^^^^^^^^^^^ move occurs because value has type `Wallet<ecdsa::signing::SigningKey<Secp256k1>>`, which does not implement the `Copy` trait

```

which is what brought me to cloning.